### ISSUE TITLE
[Feature Request] Support a grid tiling mode like Divvy / custom grid layouts

### ISSUE THREAD

**Issue by sh1mmer** [material-shell/material-shell#459]
**Is your feature request related to a problem? Please describe.**
On Mac OS X I've been using Divvy (https://mizage.com/divvy/) as a tiling manager and it has a couple of very helpful features that I think would be amazing in Material Shell: the grid layout tool, and predefined custom grid layouts.

**Describe the solution you'd like**
Specifically being able to place and size a window by dragging out across the virtual grid is amazingly helpful to create layouts. 
![image](https://user-images.githubusercontent.com/37673/98297687-a67a6b80-1f69-11eb-858e-18d998b6918f.png)

Divvy uses "shortcuts" to place things into columns easily with the keyboard.
![Screenshot from 2020-11-05 13-20-51](https://user-images.githubusercontent.com/37673/98297946-0ec94d00-1f6a-11eb-93e3-47de7b8f2eb4.png)

However I think it would be better to map to the Material Shell concepts in 2 ways:
1) allow apps in Material Shell float layout to use the grid placement / sizing
2) support a custom grid mode where the user defines slots for the apps in the space to fill. If there are less apps than the slots in the grid layout it would be better imo not to expand another app to fill that space. This allows you to leave a gap for say an adhoc terminal or file window, etc without having to reflow the rest of the layout.
3) "over-subscribed" workspace. Sometimes it's nice to have windows assigned to the same "position slot" and swap them back and forth without changing the space. For example when working in unity I have a float mode which has the unity window and the visual studio code window in the same position since I don't really look at them at the same time. Being able to assign an app to a "slot" in the grid layout would be awesome.
4) Potentially if you wanted to be really fancy I could see this being an abstraction for all layouts. E.g.:
a) break the screen for a space into a grid (e.g. 1px/grid = regular layout or 1/3rd screen res = 3 col layouts, etc )
b) define at least 1 area on the grid
c) give each defined area on the grid one of the Material Shell algorithm for how it will handle more than one app being assigned to it (e.g. sub-divide or stack)
d) defined a way to give apps in a space to each grid area

Using this approach most of the current layouts would just be one big area + the existing algorithms, but you could also then give users the choice to say split into 3 columns and have the first 2 be full columns, and the final column sub-divide horizontally, etc.

**Describe alternatives you've considered**

**Comment by PapyElGringo** [comment]
Okay I note it for our research about customizable layout

**Comment by nibirrayy** [comment]
It would be amazing to have this feature but maybe keep a keyboard centric layout like how we can use keyboard shortcut to resize windows in i3

**Comment by stukev** [comment]
+1 on predefined custom grids. I like to work in a 2x2 grid but that only works if I don't open new windows in that workspace.

**Comment by belda** [comment]
Yep a grid, where the last window, that does not fit is just stacking on a desired place, but be able which of these spots it is
